invenio_files_rest/models.py

```diff
@@ -1161,6 +1165,45 @@ def copy(self, bucket=None, key=None):
 
         return new_ob
 
+    @classmethod
+    def copy_from(cls, bucket_id, destination_bucket_id):
+        """Copy all objects from source to destination bucket.
+
+        The object versions of source bucket are copied by value i.e there is no
+        interaction with the db.
+
+        The copy operation is handled completely at the metadata level. The
+        actual data on disk is not copied. Instead, the two object versions
+        will point to the same physical file (via the same FileInstance).
+
+        All the tags associated with the current object version are copied over
+        to the new instance.
+
+        :param bucket_id: The bucket id to copy the objects from.
+        :param destination_bucket_id: The bucket id to copy the objects to.
+        :returns: A list of object versions as dictionaries.
+        """
+        new_versions = []
+
+        with db.session.begin_nested():
+            obj_columns = [col.name for col in inspect(ObjectVersion).columns]
+            for o in ObjectVersion.get_by_bucket(bucket_id):
+                new_ov = deepcopy(
+                    {
+                        key: value
+                        for key, value in o.__dict__.items()
+                        if key in obj_columns
+                    }
+                )
+                new_ov["bucket_id"] = destination_bucket_id
+                new_ov["version_id"] = uuid.uuid4()
```

Comment:
isn't uuid created on insert? or it does not work like that for bulk? just curious